### ISSUE TITLE
IE10 crashes when adding a new activity

### ISSUE THREAD

**Issue by cduicu** [bpmn-io/bpmn-js#229]
IE10 crashes when trying to add an activity to the diagram. (Windows 7 Pro + IE10)
On the same topic - Is there an official list of supported browsers ? If so what are they ?

![ie10](https://cloud.githubusercontent.com/assets/203076/6758914/3e7546a2-cf12-11e4-90c8-b0d27df45c08.png)

## Solution

The official version of bpmn-js is compatible with IE 11 only per default. Degraded support for IE 10 can be achieved though by adopting the [`diagram-js.css`](https://github.com/bpmn-io/diagram-js/blob/master/assets/diagram-js.css) as shown in [this comment](https://github.com/bpmn-io/bpmn-js/issues/229#issuecomment-337885331).

**Comment by nikku** [comment]
We are testing our toolkit against IE11/Chrome/Firefox in the modeling component and ensure the viewer is compatible with IE >= 9.

Any chance you can try to track down the issue? Maybe there is an easy fix that will allow us to support IE 10 in the modeler, too.

**Comment by vpanfilov** [comment]
@nikku, are you still intrested in IE 10 support?
I experimented today with [IE 10 virtual machine](https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/) and discovered that IE 10 crashes have something to do with some CSS styles that are applied to SVG elements during modeling operations.

Many crashes went away when I added this code to the beginning of **Canvas.prototype._updateMarker** function:
```JS
  if (element.type === 'bpmn:Process') {
    if (marker === 'connect-not-ok' || marker === 'drop-ok') return;
  }
  if (element.type === 'bpmn:SequenceFlow') {
    if (marker === 'connect-not-ok' || marker === 'drop-not-ok') return;
  }
```

**Comment by nikku** [comment]
Are these lines the only ones that break IE 10 right now?

If there is some easy fix we can fix that particular issue. I just guess there are other issues down the road, though, too.

**Comment by vpanfilov** [comment]
I haven't found other way to crash it yet. But I didn't test it fully, of course. There is also a rendering issue after initial load of diagram (not all connections are displayed). But it goes away after zooming in/out. I will update this topic if I will mess with IE10 again.

**Comment by ColinFrick** [comment]
I had the same problem, and investigated as well.

Basically using structural pseudo-classes to select path, line, polyline or polygon elements with markers crashes IE9 and IE10.

We fixed this by replacing `:nth-child` with `path` for `.djs-connection`
```css
.djs-connection.connect-ok .djs-visual > path,
.djs-connection.drop-ok .djs-visual > path {
  stroke: #90DD5F /* light-green */ !important;
}

.djs-connection.connect-not-ok .djs-visual > path,
.djs-connection.drop-not-ok .djs-visual > path {
  stroke: #E56283 /* light-red */ !important;
}
```

Related to this: https://connect.microsoft.com/IE/feedback/details/781738/ie10-crashes-with-simple-svg-css

**Comment by nikku** [comment]
@ColinFrick Thanks for looking into this and providing a solution for IE 10 users.

Unfortunately we cannot merge the workaround you've found. The reason is, that by convention the first element in a `.djs-visual` is the one that gets highlighting / coloring. Other elements that may have been added for decoration purposes remain as is and the fix for IE 10 would break that behavior.

We suggest all developers that need to support Modeling in IE 10 to apply your fix manually. I've updated the issue description with a notice concerning this matter.
